src/call.cpp

```diff
@@ -985,4 +985,46 @@ void DarlingServer::Call::S2CPerform::processCall() {
 	_sendReply(code);
 };
 
+void DarlingServer::Call::SetExecutablePath::processCall() {
+	int code = 0;
+	int fullLength = 0;
+
+	if (auto thread = _thread.lock()) {
+		if (auto process = thread->process()) {
+			std::string tmpstr;
+			tmpstr.resize(_body.buffer_size);
+			if (!process->readMemory((uintptr_t)_body.buffer, tmpstr.data(), _body.buffer_size, &code)) {
+				code = -code;
+			}
+			process->setExecutablePath(tmpstr.c_str());
+		} else {
+			code = -ESRCH;
+		}
+	} else {
+		code = -ESRCH;
+	}
+
+	_sendReply(code == 0 ? fullLength : code);
+}
+
+void DarlingServer::Call::GetExecutablePath::processCall() {
+	int code = 0;
+	uint64_t fullLength;
+
+	if (auto maybeTargetProcess = processRegistry().lookupEntryByNSID(_body.pid)) {
+		auto targetProcess = *maybeTargetProcess;
+		auto path = targetProcess->executablePath();
+		auto len = std::max(path.length() + 1, _body.buffer_size);
```

Comment:
Are you sure you want `std::max` here and not `std::min`? Otherwise, if the buffer size is bigger than the path length, we'll ending up reading garbage past the end of `path.data()`.
```suggestion
		auto len = std::min(path.length() + 1, _body.buffer_size);
```